### ISSUE TITLE
Logged events are not flushed using useStatsigClient and <StrictMode>

### ISSUE THREAD

**Issue by bgmort** [statsig-io/js-client-monorepo#11]
The following component works correctly in a React app that has a client provider using StatsigProvider higher up the tree. When [StrictMode](https://react.dev/reference/react/StrictMode) is enabled, however (which is the [default in Next.js > 13.4](https://nextjs.org/docs/pages/api-reference/next-config-js/reactStrictMode)), the log queue never gets flushed, and the events logged by the click button never get logged.

```ts
'use client'

import { useStatsigClient } from '@statsig/react-bindings'
import { useState } from 'react'

const rand = () => Math.random().toString().slice(2)

export const ClientEventsTest = () => {
  // side note: useStatsigClient() returns a new object each time, which can lead to all sorts of bugs when used directly (`client = useStatsigClient()`)
  const { client } = useStatsigClient()
  const [pageViewId] = useState(rand)
  const [clicks, setClicks] = useState(0)

  return (
    <button
      className='btn btn-primary' onClick={() => {
        const _clicks = clicks + 1
        setClicks(_clicks)
        client.logEvent('test-event', 'button-click', { random: rand(), pageViewId, clickCount: String(_clicks) })
      }}
    >Clicks: {clicks}
    </button>
  )
}
```

**Comment by tore-statsig** [comment]
Thanks for the report, we are looking into it

**Comment by daniel-statsig** [comment]
Hmm this seems related to your other report: https://github.com/statsig-io/js-client-monorepo/issues/12

Are you seeing the following warning:
```
[Statsig] Creating multiple Statsig clients with the same SDK key can lead to unexpected behavior. Multi-instance support requires different SDK keys.
```

React strict mode renders everything twice, leading to multiple StatsigClient instances. 
I think this is causing issues with the useStatsigClient hook returning the original instance which no longer has the background flush timer running as we clear it when we detect it is not the 'most recent' created instance.

**Comment by bgmort** [comment]
I'm not seeing that warning, but I can see that multiple instances are being created. I thought it was only being created once due to [React's console.log suppression](https://legacy.reactjs.org/docs/strict-mode.html#:~:text=In%20React%2017%2C%20React%20automatically,does%20not%20suppress%20any%20logs.), which I thought had been removed, but apparently the version of next that we're on is still holding on to that "feature". Incredible that useMemo doesn't actually do what it claims to do in Strict mode (at least on the version that we're on, which admittedly is out of date).

Odd that I'm not seeing the warning you mentioned, even when working around (`const realConsoleLog = window.console.log`) the log suppression issue.

**Comment by bgmort** [comment]
Ah, yes, I knew this but forgotten... `useMemo` isn't guaranteed to only run once, and so strict mode [runs it twice](https://docs.statsig.com/client/javascript-sdk/react), while suppressing the logs the second time. I'll fix my code to use a ref, and suggest you guys update your React docs to do the same.

**Comment by Mitelak** [comment]
I was also surprised that `useMemo` is suggested in the docs. The way other libraries do (e.g. [react-query](https://tanstack.com/query/latest/docs/framework/react/guides/ssr#initial-setup)) is using `useState` like:  

```tsx
 const [client] = useState(() => new StatsigClient(DEMO_CLIENT_KEY, { userID: 'a-user' }));
```

**Comment by bgmort** [comment]
@Mitelak I think that's best, and what I went with as well, although I mentioned using a ref above.

**Comment by daniel-statsig** [comment]
So I dug into it. Because strict mode is renderering twice, its causing the [shutdown method here](https://github.com/statsig-io/js-client-monorepo/blob/main/packages/react-bindings/src/StatsigProvider.tsx#L30) to be called. This is killing the background flusher in EventLogger.

Because the background flusher is currently being started during construction, there is no way to start it again once it has been shutdown.

I am working on a fix now to address this and get it all working with strict mode enabled.

**Comment by bgmort** [comment]
@daniel-statsig I found the same digging into it on Friday afternoon. Thanks for working on a fix.

**Comment by daniel-statsig** [comment]
@bgmort, I published 1.7.0 which contains a fix for this (https://github.com/statsig-io/js-client-monorepo/releases/tag/1.7.0). If you are able, can you pull it and let me know if it resolves the issue?

Note that the SDK won't auto shutdown through the StatsigProvider anymore. You may want to do something similar to this sample: https://github.com/statsig-io/js-client-monorepo/blob/1.7.0/samples/next-js/src/utils/useBootstrappedClient.tsx#L19-L28

**Comment by bgmort** [comment]
@daniel-statsig I tried but it didn't work for me. In your `useBootstrappedClient` pattern, the init happens in a useEffect, the values aren't set on the server render or the initial client render, so I get all false values on the initial render.

I was able to get it to work with some tweaks, but it's fairly messy. We're considering turning off Next's strict mode option and inserting the StrictMode component ourselves below the providers in the tree.
